### ISSUE TITLE
Make e2e tests rerunnable (spec 8)

### ISSUE THREAD

**Issue by mehansen** [CDCgov/prime-simplereport#7011]
### Description
Many of the specs in our end-to-end Cypress tests rely on data set up by the other specs. This makes local debugging tricky because you have to reset the data and run every spec in order before the one you're trying to test. It also makes it harder to tell what's wrong when a spec fails because it will usually cause failures in many following test cases. We want to fix this up to the tests easier to work with. This would also set us up to potentially be able to explore running these tests in parallel.

### Proposed solution 
Use before and after hooks to set up data for each end-to-end spec and then clean it up after testing is complete.
See if there's tooling to help with this.

### Acceptance criteria
Each spec can be run and rerun individually on its own. All prereqs for the test cases to run are taken care of within the spec and any data created is cleaned up after.

### Additional notes
See work on #6465 for an example of how to do this

**Comment by mpbrown** [comment]
For the purposes of this ticket, spec 8 is already rerunnable and isolated from the other tests. This was demonstrated in [this GitHub testing workflow log which shows spec 8 running at the very beginning of the e2e test order](https://github.com/CDCgov/prime-simplereport/actions/runs/8189799260/job/22399761306#step:6:130) followed by a duplicate of spec 8. In the branch running this workflow, spec 8 was renamed to 01a and 01b and spec 1 was renamed to spec 11.

Spec 8 still proves difficult to get to run locally outside of docker. Although it is possible to get a wiremock instance to run locally and have the mocked Okta responses for the test correctly return using wiremock, there remains an issue with persisting sessions when running the backend locally.

When spec 8 runs `cy.visit("/uac/?activationToken=NOr20VqF5M6m8AnwcSUJ");`, there are three relevant API calls involved in AccountCreationApp.tsx. 

- First it calls `/user-account/user-status` endpoint and passes in the activation token. The backend in `LiveOktaAuthentication.getUserStatus` will then interpret that as the user not being activated yet and correctly return a status of PENDING_ACTIVATION. 
- Next, it calls the `/user-account/initialize` endpoint and passes in the activation token. The backend in `LiveOktaAuthentication.activateUser` will then make a post request to "Okta" in order to activate that user. The Okta url in this case actually points to wiremock which will match the request to a mocked response and return that. In this case, the userId returned from `activateUser` is correctly the same userId from the mocked response. This userId then gets set as a session attribute via `request.getSession().setAttribute(USER_ID_KEY, userId)` using Spring's `HttpServletRequest`
- Finally, the component calls `/user-account/user-status` again - this time without the activation token - and expects that the userId that was just set in the previous request will be available on the session. Unfortunately, the session returned by `request.getSession()` has a different id than the previous session so the userId attribute is not available. This then leads to the status being returned as unknown which causes a 404 error in the test.
